### PR TITLE
feat: add new plots (heatmap, correlation matrix, raincloud) and refactor general vis

### PR DESCRIPTION
**Pull request by dvmoritzschoefl**
Since it is not straight forward to test the dependency chain of the open vis PRs, I merged the other ones in this PR. This includes

https://github.com/datavisyn/visyn_core/pull/7
https://github.com/datavisyn/visyn_core/pull/49
https://github.com/datavisyn/visyn_core/pull/54
https://github.com/datavisyn/visyn_core/pull/60
https://github.com/datavisyn/visyn_core/pull/62
https://github.com/datavisyn/visyn_core/pull/55

### CODE REVIEW

**Review by dvchristianbors — CHANGES_REQUESTED** [review]
I summarize my observations on the new vis here:

#### Multiple vis components
Bar plot, sankey plot, hexbin plot:
* Can we add categorical colors in the props so that we can match colors between lineup and the general vis (this already used to work for bar charts, and is working for scatter plot)?

#### Raincloud plot:
* The boxplot whiskers are inconsistent with the violin plot. How is it defined? min-max? It looks like the violin plot's boxplot spans the IQR.
  image
  Screenshot 2023-08-25 102414
* With some score columns, the app breaks due to a `querySelector` error (EDIT: I think this is only due to some Ordino related adjustments that still need to be made)
  Screenshot 2023-08-25 095157



#### Sankey plot
* instructions are not styled
  image
  Use an error/warning boundary, similar to scatter plot warning
  image

  

* plotly default (?) icons show in the canvas
  image

#### Correlation plot
* Sometimes a rogue correlation dot appears after removing a value column
  https://github.com/datavisyn/visyn_core/assets/84898946/8ae49590-eac1-4a2f-ab2d-828ffedb7414

**Comment by dvdanielamoitzi** [comment]
We're using the following interfaces / enums which are not exported anymore:
EBarDirection, EBarDisplayType, EBarGroupingType, IBarConfig. We defenitely need IBarConfig

**Comment by dvchristianbors** [comment]
> We're using the following interfaces / enums which are not exported anymore: EBarDirection, EBarDisplayType, EBarGroupingType, IBarConfig. We defenitely need IBarConfig

In Reprovisyn, we are using all specific vis configs. If this is not recommended, please let us know what to change in reprovisyn.

**Comment by thinkh** [comment]
@dvmoritzschoefl Thanks for the latest refactoring and fixing the Ordino integration. Regarding the status of this PR: Did you address the [Bob's feedback](https://github.com/datavisyn/visyn_core/pull/76#pullrequestreview-1595266608) already?

I still see the plotly toolbar in the sankey diagram and in addition to that I would like to zoom and pan in this plot. Could you add this?

**Comment by dvmartinweigl** [comment]
Could you please limit the x-axis label length and add a `text-overflow: ellipsis`
Otherwise the actual plot area becomes really small (see the screenshot)  

image

**Comment by dvmoritzschoefl** [comment]
> Could you please limit the x-axis label length and add a `text-overflow: ellipsis` Otherwise the actual plot area becomes really small (see the screenshot)
> 
> image

I do not see an easy way to do this. I also think this is out of scope for this PR which is only there to combine the existing vis into one framework. (Except the bug was introduced in this PR of course)

**Comment by dvmoritzschoefl** [comment]
> #### Some other bugs I found while testing your branch in bioinisght:
> * For the column selection (numerical and categorical) we had a tool tip with `name` + `description` before. Now it is missing
> * The y-axis labesl of the bar chart should all be left aligned
>   Screenshot from 2023-09-21 14-41-30
> * The hexbin plot does not resize vertically when i reduce the mosaic height in bioinisight
>   Screenshot from 2023-09-21 14-43-39

the default for most vis frameworks is to right align items on the y axis, which is what makes most sense. what you see here is an artefact of the ellipsis which is hardcoded in the barchart (and not responsive like in plotly)

as for the hexbin plot yes, it does not seem to resize at all when the area is shrinking i will take a look at that

File: tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "visyn_scripts/config/tsconfig.template.json",
   "compilerOptions": {
+    "jsx": "react",
```
Comment (thinkh) [inline tsconfig.json:4]:
```suggestion
```

Can we remove this line? It is already the default in visyn_scripts:

https://github.com/datavisyn/visyn_scripts/blob/0d633022e00dd78c26da00e1b7716cff3a6ba828/config/tsconfig.template.json#L10C5-L10C6

File: package.json [hunk 2]
```diff
@@ -150,16 +151,21 @@
     "@types/react": "^18.2.0",
     "@types/react-dom": "^18.2.0",
     "@types/react-plotly.js": "^2.5.0",
+    "arquero": "^5.2.0",
+    "chromatic": "^6.19.9",
```
Comment (thinkh) [inline package.json:155]: I think chromatic should be moved to the `devDependencies`. Do you agree?

**Comment by dvdanielamoitzi** [comment]
In the previous version we had a tooltip for categorical / numerical columns that we used in BioInsight to display more information on the selected items:
image

**Comment by dvchristianbors** [comment]
> In the previous version we had a tooltip for categorical / numerical columns that we used in BioInsight to display more information on the selected items: image

@dvdanielamoitzi  I'm a bit confused, did this use to work but it does not work any more? I just tried this in Reprovisyn, and the tooltip works as expected:
image

**Review by dvchristianbors — APPROVED** [review]
The PR now looks good to me. I tested it with Ordino Enterprise and could not find any critical issues.

**Review by dvmartinweigl — APPROVED** [review]
Tested all the vis plots in Bioinisght and found no critical bugs. So this one is ready to merge from my perspective.

**Pull request merged**